osu.Game.Rulesets.Osu.Tests/Editor/TestSceneObjectMerging.cs

```diff
@@ -168,6 +168,99 @@ public void TestNonMerge()
+
+            AddAssert("slider created", () => circle1 is not null && circle2 is not null && sliderCreatedFor(
+                (pos: circle1.Position, pathType: PathType.Linear),
+                (pos: circle2.Position, pathType: null)));
+
+            AddStep("start editor clock", () =>
+            {
+                EditorClock.Seek(sliderStartTime);
+                EditorClock.Start();
+            });
+
+            AddStep("stop editor clock", () =>
+            {
+                EditorClock.Stop();
+            });
+
+            AddStep("undo", () => Editor.Undo());
+            AddAssert("merged objects restored", () => circle1 is not null && circle2 is not null && objectsRestored(circle1, circle2));
+        }
+
+        [Test]
+        public void TestMergeCircleSliderHitsound()
+        {
+            HitCircle? circle1 = null;
+            Slider? slider = null;
+            HitCircle? circle2 = null;
+            double sliderStartTime = 0;
+
+            AddStep("select a circle, slider, circle", () =>
+            {
+                circle1 = (HitCircle)EditorBeatmap.HitObjects.First(h => h is HitCircle);
+                slider = (Slider)EditorBeatmap.HitObjects.First(h => h is Slider && h.StartTime > circle1.StartTime);
+                circle2 = (HitCircle)EditorBeatmap.HitObjects.First(h => h is HitCircle && h.StartTime > slider.StartTime);
+                EditorClock.Seek(circle1.StartTime);
+                EditorBeatmap.SelectedHitObjects.Add(circle1);
+                EditorBeatmap.SelectedHitObjects.Add(slider);
+                EditorBeatmap.SelectedHitObjects.Add(circle2);
+                sliderStartTime = circle1.StartTime;
+            });
+
+            mergeSelection();
+
+            AddAssert("slider created", () =>
+            {
+                if (circle1 is null || circle2 is null || slider is null)
+                    return false;
+
+                var controlPoints = slider.Path.ControlPoints;
+                (Vector2, PathType?)[] args = new (Vector2, PathType?)[controlPoints.Count + 2];
+                args[0] = (circle1.Position, PathType.Linear);
+
+                for (int i = 0; i < controlPoints.Count; i++)
+                {
+                    args[i + 1] = (controlPoints[i].Position + slider.Position, i == controlPoints.Count - 1 ? PathType.Linear : controlPoints[i].Type);
+                }
+
+                args[^1] = (circle2.Position, null);
+                return sliderCreatedFor(args);
+            });
+
+            AddStep("start editor clock", () =>
+            {
+                EditorClock.Seek(sliderStartTime);
+                EditorClock.Start();
+            });
+
+            AddStep("stop editor clock", () =>
+            {
+                EditorClock.Stop();
+            });
+
+            AddStep("undo", () => Editor.Undo());
+            AddAssert("merged objects restored", () => circle1 is not null && circle2 is not null && objectsRestored(circle1, circle2));
```

Comment:
What's going on here? Shouldn't you be checking `Samples` to make sure they are populated and correct?

Reply:
Edited the tests to test if `Samples[0]` exist.